### PR TITLE
Mean miner

### PR DESCRIPTION
**Pull request by lucafavatella**
Finishes [PT-153051503](https://www.pivotaltracker.com/story/show/153051503) and [PT-152915859](https://www.pivotaltracker.com/story/show/152915859).

* [x] Depends on https://github.com/tromp/cuckoo/pull/33 to be reviewed upstream first. (UPDATE: merged)
* [x] Depends on smoke test on Linux environment. (UPDATE: tested on dev1, mines stably and faster than lean.)

### CODE REVIEW

**Comment by lucafavatella** [comment]
I force-pushed considering upstream cuckoo merge and binary renames. (And I rebased on master.)

File: apps/aecuckoo/Makefile [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+	priv/bin/mean30s-generic \
+	priv/bin/mean28s-generic \
+	priv/bin/mean16s-generic \
+	priv/bin/mean30s-avx2 \
+	priv/bin/mean28s-avx2 \
+	priv/bin/mean16s-avx2 \
+	priv/bin/lean30 \
 	priv/bin/lean28 \
+	priv/bin/lean16 \
+	priv/bin/verify30 \
```
Comment (hanssv) [inline apps/aecuckoo/Makefile]: Except for the "smoke test" is there a point in including the verifiers? We don't use them anymore I realise...

Comment (lucafavatella) [reply apps/aecuckoo/Makefile]: I think it is useful from a user or operator perspective to have a consistent set of upstream binaries so command-line checks can be performed. I can remove them if that is the agreement though...

**Comment by lucafavatella** [comment]
I see two reviewers approved. I see CI passed.

**Pull request merged**